server/src/main/java/org/candlepin/model/Product.java

```diff
@@ -104,10 +104,16 @@
     @LazyCollection(LazyCollectionOption.EXTRA) // allows .size() without loading all data
     private List<ProductContent> productContent;
 
+    /*
+     * hibernate persists empty set as null, and tries to fetch
+     * dependentProductIds upon a fetch when we lazy load. to fix this, we eager
+     * fetch.
+     */
     @ElementCollection
     @CollectionTable(name = "cpo_product_dependent_products",
                      joinColumns = @JoinColumn(name = "product_uuid"))
     @Column(name = "element")
+    @LazyCollection(LazyCollectionOption.FALSE)
```

Comment:
I'm curious what the scenario is here that caused a problem. In all of the constructors and in the setter, we set the collection to an empty hash set, so there aren't many cases where this can be null.